code/lib/telemetry/src/package-json.ts

```diff
@@ -28,10 +28,8 @@ export const getActualPackageJson = async (packageName: string) => {
   return packageJson;
 };
 
-// Note that this probably doesn't work in PNPM mode
+// Note that this probably doesn't work in Yarn PNP mode because @storybook/telemetry doesn't depend on @storybook/cli
 export const getStorybookCoreVersion = async () => {
-  const coreVersions = await Promise.all(
-    ['@storybook/core-common', '@storybook/core-server'].map(getActualPackageVersion)
-  );
-  return coreVersions.find((v) => v.version)?.version;
+  const { version } = await getActualPackageVersion('@storybook/cli');
+  return version;
```

Comment:
can we add a backup in that case?

Reply:
I'm not sure how? Presumably you can't `require.resolve` anything in PnP mode, because you're running with `npx` so you don't depend on anything in the project, and thus you can't resolve it.

Reply:
To be clear, this didn't work before either in PnP, it never worked. Confirmed on a call that we're then OK to merge.